Adding tax observer (#1011)
ihmeuw/vivarium_research: docs/source/concept_models/vivarium_census_synthdata/concept_model.rst

- Includes mail/online, telephone, and personal visits 

Taxes
^^^^^

Taxes, as we all know, can contain many different forms and processes. 
For this model, we will split the tax information into two main sections: 
W2/1099 forms from employers; and 1040 forms from simulants. We will look 
at these separately, starting with W2 and 1099 forms. 

W2 and 1099 Forms
'''''''''''''''''

**When to Sample** 

- Sample compiled on the time step containing Jan 1st of each year (the time step might end on Jan 2nd, Jan 15th, Jan 27th, etc.)

> - However, we will want to track every job a simulant has had for any time step within a calendar year, which might require additional observers. If a simulant changes jobs in March of 2020, their tax documents on Jan 1st of 2021 will need to include both their current job, and their job from February of 2020. 

Reviewer: should we track jobs within the last calendar year ending on Jan 1st, or ending on the last day of the timestep containing Jan 1st?

Author: Good callout - I added clarification. It can be through the time step containing Jan 1st

**What to Sample** 

.. list-table:: Simulant Attribute to Sample 
  :widths: 20
  :header-rows: 0

  * - Unique simulant ID (for PRL tracking)
  * - First name
  * - Middle initial 
  * - Last name
  * - Age 
  * - DOB 
  * - Mailing Address 
  * - Social Security Number 

>   * - Income 

Reviewer: sidenote: we need docs describing an income column (we currently agreed on slack that we'll assign the exact same income to everyone who is employed, and zero to everyone else)

Author: Yes, I misunderstood. I thought income was randomly assigned, so people would have higher and lower incomes even if the numbers were random. I think income will be built out more in the future, so I want to leave these pieces in the docs, but I will clarify that the income does not apply for now.

  * - Employer ID 
  * - Employer Address 
  * - Employer Zip Code 
  * - Type of Tax Form (W2 or 1099)

**Who to Sample** 

Everyone who has had an employer listed within the current calendar year 
will receive either a W2 or a 1099 form. For those with multiple jobs during 
the year, they will be duplicated and receive multiple forms. 

The rate of the the types of forms are below. This data is 
from a review of 2016 tax data by [Lim_2019]_ 

.. list-table:: Percent W2 versus 1099  
  :widths: 10 10 
  :header-rows: 1

>   * - Form Type 

Reviewer: do you want there to be any persistence from year to year as to whether someone is filing a 1099 or a w2?

Author: For now no, we can do randomly. Since we don't use that information it shouldn't matter. I will clarify in the docs.

    - Percent Receiving 
  * - W2 
    - 94.65% 
  * - 1099 
    - 5.35%

**Data Errors/Noise** 
In the future, we will add a noise function designed to replicate 
missed or incorrect data. This includes incorrect name spelling, 
addresses, age, and person swaps. 

Some additional tax specific errors include: SSN randomly being filled 
for those who are missing one, mailing address being different than the 

> home address, and issues with employer names, addresses, or IDs. 

Reviewer: note: we currently only have one address --- but we could add a separate "mailing" vs "home" address

Author: Thanks for the note - I will add this as a limitation/future add.

#. There are some employed people who do not receive a W2 or 1099, often for "under the table" work. This phenomenon might be easiest to include in the simulation as these individuals would not have a listed employer despite having an income. I chose to have all those that have an employer listed receive a W2/1099. 
#. Many workers might have multiple jobs simultaneously and receive multiple forms. This is not included in the current model. 
#. Elderly people can still have to file taxes based on social security payments, but would likely not have an employer in our model. 

1040 Form
'''''''''

**When to Sample** 

- Sample compiled on the time step containing April 15th of each year 
- Sample on a single time step for now 

**What to Sample** 

.. list-table:: Simulant Attribute to Sample 
  :widths: 20 20 
  :header-rows: 0

  * - Unique simulant ID (for PRL tracking)
    - Notes 
  * - First name
    - 
  * - Middle initial 
    - 
  * - Last name
    - 
  * - Age
    -  
  * - DOB
    -  
  * - Mailing Address
    -  
  * - Social Security Number 
    -
  * - Income 
    -

>   * - Employer ID

Reviewer: if someone has multiple employers, would we also have multiple income rows? would you want/need to know when in the last year they changed employers?

    - Can have multiple rows if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer Address 
    - Can have multiple rows if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer Zip Code 
    - Can have multiple rows if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  

>   * - Joint Filer or Dependent 

Reviewer: if only one form, how would you like to choose who submits the form?

>   * - Joint Filer or Dependent 

Reviewer: actually --- i have only ever filed single, and don't know what filing jointly looks like, so this might be naive, but --- if filing jointly, do all of those identification columns get duplicated? like person_1 name, person_2 name, person_1 employer_id, person_2 employer_id, etc.?

>   * - Joint Filer or Dependent 

Author: Honestly, I have never filed jointly either! I have expanded the table a bit to try and make this clearer. I believe yes, for the joint filing you would just include all of the information for the spouse - name, employer info, etc. I separated our dependents to try and make this more clear. 

Only one simulant would submit the form, the other is just included as a joint filer. So the reference person would submit the form, and the spouse would not submit a form. I added this as well.

    - Identify the below simulant as a joint filer or a dependent. Can have multiple rows if multiple simulants are included on the tax documents (e.g., for a simulant with a spouse and 2 children, there will be 1 joint filer and 2 dependents) 
  * - First name 
    - 
  * - Middle initial 
    - 
  * - Last name 
    - 
  * - Age in years  
    - 

**Who to Sample** 

Not everyone who receives a W2 or 1099 will end up filing taxes. 
However, those who do not are concentrated in low incomes for whom 
taxes are not required. Currently, we will chose to have all those 
who are legally required to file taxes, file taxes. This is a 
limitation and is listed below. 

> For simulants that receive below the minimum income, 42.14% will 

Reviewer: should we choose 42.14% uniformly at random from all simulants who had an employer in the last year, and with less than the minimum income?

should there be persistence from year to year, as to who files vs. doesn't?

Author: Yep, we can just randomly select some folks to not file. We don't need persistence, I will clarify in the docs.

still file taxes. [Cilke_1998]_ The remainder will not. The minimum 
income is based on the household structure and if listed in the table below. 

.. list-table:: Minimum Income  
  :widths: 20 20 
  :header-rows: 0

  * - Simulant Type 
    - Minimum Income 
  * - Single filing, under 65 
    - $12,550 
  * - Single filing, over 65 
    - $14,250
  * - Married joint filing, both under 65 
    - $25,100
  * - Married joint filing, one under 65 
    - $26,450
  * - Married joint filing, both over 65 
    - $27,800
  * - Married separate filing 

>     - $5

Reviewer: is this a typo?

Author: Fair question but no, this is real!

Reviewer: wow that's fascinating; is that intended as some sort of incentive to file jointly if you're married with low income?

Based on the household structure, the following rules can be applied 
for who files taxes: 

> - Assume that 95% of spouses file jointly, this can be randomly assigned. [Nolo]_ Others will file separately. 

Reviewer: we have the following "relationship to household head" labels for simulants:
    20: "Reference person",
    21: "Opp-sex spouse",
    22: "Opp-sex partner",
    23: "Same-sex spouse",
    24: "Same-sex partner",
    25: "Biological child",
    26: "Adopted child",
    27: "Stepchild",
    28: "Sibling",
    29: "Parent",
    30: "Grandchild",
    31: "Parent-in-law",
    32: "Child-in-law",
    33: "Other relative",
    34: "Roommate",
    35: "Foster child",
    36: "Other nonrelative",
    37: "Institutionalized GQ pop",
    38: "Noninstitutionalized GQ pop",

I assume but want to confirm --- the set of all "spouses" in the simulation on any given timestep is the set of all pairs of [Reference person, Opp-sex spouse], [Reference person, Same-sex spouse] in households, ignoring if there are, for example: two parents in a household, two "other nonrelatives" (who conceivably could be married) in a household, etc. ?

^ and from that set of people we choose 95% of them to file jointly?

Reviewer: also: do you want there to be year-to-year persistence in who files jointly?

Reviewer: ^ if so, do you want us to do anything to ensure it's always about 95% of spouses year to year that are filing jointly?

Author: Yes, the only spouses we will cover are [Reference person, Opp-sex spouse] and [Reference person, Same-sex spouse]. I agree there could be other married couples within the house, our current system doesn't work particularly well for complex family structures. I will clarify in the docs. 

I think lets do random for filing jointly as I imagine that is simpler. Again, I will add to the docs.

- All other non-married simulants in a household with a W2 or 1099 will file separately, based on the income rules above (e.g., a low-income earner in a house with other earners will be randomly assigned to file or not file independent of others in the household) 
- If the reference person in the household is employed and filing taxes, they will claim all in the house that qualify as a dependent based on these rules: 
    * Do not claim a "housemate/roommate" or "other nonrelative" as a dependent 
    * The dependent must have lived with the reference person for greater than 1 year 
    * The dependent's income must be below $4300 

> - If the reference person is unemployed or not filing taxes (low income), the highest earner in the household will claim dependents based on the same rules 

Reviewer: if the highest earner is a housemate/roommate or "other nonrelative", does this still apply? what if everyone in the house is an "other nonrelative"?

Author: Good questions! I have added to the docs to have it be relatives only.

- Sample compiled on the time step containing Jan 1st of each year (the time step might end on Jan 2nd, Jan 15th, Jan 27th, etc.)
- However, we will want to track every job a simulant has had for any time step within a calendar year, which might require additional observers. If a simulant changes jobs in March of 2020, their tax documents on Jan 1st of 2021 will need to include both their current job, and their job from February of 2020. Jobs can be tracked for complete time steps, so through the last day of the time step containing Jan 1st. 

**What to Sample** 

.. list-table:: Simulant Attribute to Sample 
  :widths: 20
  :header-rows: 0

  * - Unique simulant ID (for PRL tracking)
  * - First name
  * - Middle initial 
  * - Last name
  * - Age 
  * - DOB 
  * - Mailing Address 
  * - Social Security Number 
  * - Income 
  * - Employer ID 
  * - Employer Address 
  * - Employer Zip Code 
  * - Type of Tax Form (W2 or 1099)

**Who to Sample** 

Everyone who has had an employer listed within the current calendar year 
will receive either a W2 or a 1099 form. For those with multiple jobs during 
the year, they will be duplicated and receive multiple forms. We currently 

> will not model persistance from year to year on which type of form. 

Reviewer 2: ```suggestion
will not model persistence from year to year on which type of form. 
```

The rate of the the types of forms are below. This data is 

> from a review of 2016 tax data by [Lim_2019]_ 

Reviewer 2: ```suggestion
from a review of 2016 tax data by [Lim_2019]_.
```

#. 1099 forms are often used by self-employed people or those with small businesses. These can contain errors related in employer information. 
#. There are some employed people who do not receive a W2 or 1099, often for "under the table" work. This phenomenon might be easiest to include in the simulation as these individuals would not have a listed employer despite having an income. I chose to have all those that have an employer listed receive a W2/1099. 
#. Many workers might have multiple jobs simultaneously and receive multiple forms. This is not included in the current model. 
#. Elderly people can still have to file taxes based on social security payments, but would likely not have an employer in our model. 
#. Currently mailing addresses are the same as home addresses. This is not true, especially for rural populations. We plan to add this to the model later. 

1040 Form
'''''''''

**When to Sample** 

- Sample compiled on the time step containing April 15th of each year 
- Sample on a single time step for now 

**What to Sample** 

.. list-table:: Simulant Attribute to Sample 
  :widths: 20 20 
  :header-rows: 0

  * - Unique simulant ID (for PRL tracking)

>     - Notes 

Reviewer 2: Was this intended to have a header row?

Author: Wow, great catch!

  * - First name
    - 
  * - Middle initial 
    - 
  * - Last name
    - 
  * - Age
    -  
  * - DOB
    -  
  * - Mailing Address
    -  
  * - Social Security Number 
    -
  * - Income 
    - Can have multiple rows if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer ID
    - Can have multiple rows if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer Address 
    - Can have multiple rows if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer Zip Code 
    - Can have multiple rows if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Joint Filer 

>     - This row through 'dependent filer' are to be included if there is a joint filer ONLY 

Reviewer 2: ```suggestion
    - This row through 'dependent' are to be included if there is a joint filer ONLY 
```

    -  
  * - Social Security Number 
    -
  * - Income 
    - Can have multiple rows if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer ID
    - Can have multiple rows if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer Address 
    - Can have multiple rows if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer Zip Code 
    - Can have multiple rows if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Dependent
    - This row through the end are to be included for each dependent on the tax filing 
  * - First name 
    - 
  * - Middle initial 
    - 
  * - Last name 
    - 
  * - Age 
    -  

**Who to Sample** 

Not everyone who receives a W2 or 1099 will end up filing taxes. 
However, those who do not are concentrated in low incomes for whom 
taxes are not required. Currently, we will chose to have all those 
who are legally required to file taxes, file taxes. This is a 
limitation and is listed below. 

For simulants that receive below the minimum income, 42.14% will 
still file taxes. [Cilke_1998]_ The remainder will not. The minimum 

> income is based on the household structure and if listed in the table below. 

Reviewer 3: ```suggestion
income is based on the household structure and is listed in the table below. 
```

> We will not model persistance year to year. 

Reviewer 2: ```suggestion
We will not model persistence year to year. 
```

**In the current model, no one will be low income, this will be changed later.** 

.. list-table:: Minimum Income  
  :widths: 20 20 
  :header-rows: 0

  * - Simulant Type 
    - Minimum Income 
  * - Single filing, under 65 
    - $12,550 
  * - Single filing, over 65 
    - $14,250
  * - Married joint filing, both under 65 
    - $25,100
  * - Married joint filing, one under 65 
    - $26,450
  * - Married joint filing, both over 65 
    - $27,800
  * - Married separate filing 
    - $5

Based on the household structure, the following rules can be applied 
for who files taxes: 

- Assume that 95% of spouses file jointly, this can be randomly assigned. [Nolo]_ Others will file separately. 
    * The only spouses we will recognize are [Reference person, Opp-sex spouse] and [Reference person, Same-sex spouse]. 
    * The reference person will submit the form, the spouse will be listed as the joint filer. 

>     * There does not need to be persistance in who files jointly, it can be re-randomly drawn each year. 

Reviewer 2: ```suggestion
    * There does not need to be persistence in who files jointly, it can be randomly re-drawn each year. 
```

> - All other non-married simulants in a household with a W2 or 1099 will file separately, based on the income rules above (e.g., a low-income earner in a house with other earners will be randomly assigned to file or not file independent of others in the household) 

Reviewer 2: ```suggestion
- All other non-married simulants in a household with a W2 or 1099 will file separately, based on the income rules above (e.g., a low-income earner in a house with other earners will be randomly assigned to file or not file, independent of others in the household) 
```

I first read this as whether or not they would file independently, i.e. as a dependent or not.

Reviewer 2: Also, it is not clear to me how this interacts with claiming dependents. If someone has a W2 but is below the $4,300 number for being a dependent (and also meets the other criteria for being a dependent), what happens?

Author: I added something to clarify - but I think the confusion is that you can BOTH file taxes for yourself AND be claimed as a dependent. Imagine a high schooler who worked a summer job. They likely would get some money back if they file taxes, but also can be claimed as a dependent. I am basically assuming that everyone maximizes their tax benefits and so is claimed as a dependent if they can be, but some also will choose to file their own taxes.

home address, and issues with employer names, addresses, or IDs. 

These are not currently modeled. 

**Limitations and Possible Future Adds** 

#. Sampling on a single time step is not representative of how tax documents are compiled. 
#. Errors are made in W2s and 1099s by companies frequently, due to employees moving or changing information without communicating changes. These W2s can be reissued which leads to duplicates, or employees might not adjust them leading to different information between W2/1099s and 1040 forms. This is not currently modeled. 
#. 1099 forms are often used by self-employed people or those with small businesses. These can contain errors related in employer information. 
#. There are some employed people who do not receive a W2 or 1099, often for "under the table" work. This phenomenon might be easiest to include in the simulation as these individuals would not have a listed employer despite having an income. I chose to have all those that have an employer listed receive a W2/1099. 
#. Many workers might have multiple jobs simultaneously and receive multiple forms. This is not included in the current model. 

> #. Elderly people can still have to file taxes based on social security payments, but would likely not have an employer in our model. 

Reviewer: side note --- we currently give everyone over "working age" an equal likelihood of being employed, and do not have any sort of retirement implemented.

Author: Thanks! This is on my list to edit now :)

- If the reference person in the household is employed and filing taxes, they will claim all in the house that qualify as a dependent based on these rules: 
    * Do not claim a "housemate/roommate" or "other nonrelative" as a dependent 
    * The dependent must have lived with the reference person for greater than 1 year 
    * The dependent's income must be below $4300 
- If the reference person is unemployed or not filing taxes (low income), the relative (not housemate or other non-relative) who is the highest earner in the household will claim dependents based on the same rules. If there are no relatives filing taxes in the household, no one will claim dependents. **For now when incomes are all the same, randomly assign to any relative earner**

**Data Errors/Noise** 
In the future, we will add a noise function designed to replicate 
missed or incorrect data. This includes incorrect name spelling, 
addresses, age, and person swaps. 

> Some additional tax specific errors include: SSN randomly being filled 

Reviewer: I think but am clarifying: this type of noise will be a future include / is not to be implemented now, right?

Author: Yes! This is a future add

Taxes
^^^^^

Taxes, as we all know, can contain many different forms and processes. 
For this model, we will split the tax information into two main sections: 
W2/1099 forms from employers; and 1040 forms from simulants. We will look 
at these separately, starting with W2 and 1099 forms. 

W2 and 1099 Forms
'''''''''''''''''

**When to Sample** 

- Sample compiled on the time step containing Jan 1st of each year (the time step might end on Jan 2nd, Jan 15th, Jan 27th, etc.)
- However, we will want to track every job a simulant has had for any time step within a calendar year, which might require additional observers. If a simulant changes jobs in March of 2020, their tax documents on Jan 1st of 2021 will need to include both their current job, and their job from February of 2020. Jobs can be tracked for complete time steps, so through the last day of the time step containing Jan 1st. 

**What to Sample** 

.. list-table:: Simulant Attribute to Sample 
  :widths: 20
  :header-rows: 0

  * - Unique simulant ID (for PRL tracking)
  * - First name
  * - Middle initial 
  * - Last name
  * - Age 
  * - DOB 
  * - Mailing Address 
  * - Social Security Number 

>   * - Wages (income from this job)

Reviewer: is "wages" a distinct concept from income? if yes, could you define it somewhere  nearby --- if no, could we call this 'income'?

Author: The census asked for this to be "wages" since it is from the companies point of view. So it is "how much WE payed YOU" not "how much I got paid". I added an explanation to the docs though!

  * - Employer ID 
  * - Employer Name 
  * - Employer Address 
  * - Employer Zip Code 
  * - Type of Tax Form (W2 or 1099)

If a simulant does not have a social security number but is 
employed, they will need this number to be filled in. If there 

> is a person in their household who has a SSN, use this number 

Reviewer: which person in there household do we choose? uniform at random? some specific rule?

Author: Added this - can choose at random. Thanks!

instead. If there is not a person in their household with a SSN 
then fill in a random number. This is designed to reflect 
undocumented immigrants who might use fake or no longer 
valid SSNs to obtain employment. 

For this observer, a new row should be made for each **employment**, not 
each simulant. This means that a simulant can have multiple rows of 
data, or just one row of data. 

Here is an example: 

.. image:: W2_example.PNG

**Who to Sample** 

Everyone who has had an employer listed within the current calendar year 
will receive either a W2 or a 1099 form. For those with multiple jobs during 
the year, they will be duplicated and receive multiple forms. We currently 
will not model persistence from year to year on which type of form. 

The rate of the the types of forms are below. This data is 
from a review of 2016 tax data by [Lim_2019]_ . 

> .. list-table:: Percent W2 versus 1099  

Reviewer: i think this is probably per job, but just to make sure: is this percent per job, or percent per person?

i.e., if someone had two jobs, do we select if they file a w2 vs 1099 for their first job with 94.65 v. 5.35 % probability, and then independently sample again for their second job? or do we sample once, and then all of one persons jobs return the same type of form?

Author: Added, thanks! We can select per job.

    -
  * - Income 
    - Can have multiple columns if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer ID
    - Can have multiple columns if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer Name
    - Can have multiple columns if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer Address 
    - Can have multiple columns if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Employer Zip Code 
    - Can have multiple columns if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Type of tax form (W2 or 1099)
    - Can have multiple columns if simulant has multiple jobs in the prior year (multiple W2/1099 forms)  
  * - Dependent
    - This columns through the end are to be included for each dependent on the tax filing 
  * - First name 
    - 
  * - Middle initial 
    - 
  * - Last name 
    - 
  * - Age 
    -  
  * - Social Security Number or ITIN
    -

If a simulant does not have a SSN but is filing taxes, please 
include an Individual Taxpayer Identification Number (ITIN) instead. 

> This is a 9 digit number that starts with 

Reviewer: should ITN be unique?

Author: I don't think it matters right now. Thanks!

income is based on the household structure and is listed in the table below. 
We will not model persistence year to year. 

**In the current model, no one will be low income, this will be changed later.** 

.. list-table:: Minimum Income  
  :widths: 20 20 
  :header-rows: 0

  * - Simulant Type 
    - Minimum Income 
  * - Single filing, under 65 
    - $12,550 
  * - Single filing, over 65 
    - $14,250
  * - Married joint filing, both under 65 
    - $25,100
  * - Married joint filing, one under 65 
    - $26,450
  * - Married joint filing, both over 65 
    - $27,800
  * - Married separate filing 
    - $5

Based on the household structure, the following rules can be applied 
for who files taxes: 

- Assume that 95% of spouses file jointly, this can be randomly assigned. [Nolo]_ Others will file separately. 
    * The only spouses we will recognize are [Reference person, Opp-sex spouse] and [Reference person, Same-sex spouse]. 
    * The reference person will submit the form, the spouse will be listed as the joint filer. 
    * There does not need to be persistence in who files jointly, it can be re-drawn each year. 

> - All other non-married simulants in a household with a W2 or 1099 will file separately, based on the income rules above (e.g., a low-income earner in a house with other earners will be randomly assigned to file or not file, independent of others in the household). Please note that simulants can BOTH be claimed as a dependent AND file their own taxes. 

Reviewer: just to make sure i'm following: 

- if person A can claim person B as a dependent, they always will.
- if person B had an employer + income in the past year, the only reason they won't submit a w2 or 1099 and a 1044 is if they have a sufficiently low income, and they are selected as among the 67% who still file taxes; is that correct? i.e., there aren't other reasons someone who is claimed as a dependent might not file taxes?

Author: Yes - this is correct! Anyone who can be claimed as a dependent will be claimed. Regardless of if you are claimed, you still decide to do or not do taxes independently. 

In real life, there is a whole "informal economy" where people have jobs that pay in cash and don't show in taxes. But we are not modeling this at all.

- All simulants eligible to be dependents will be assigned to a relative within the household 
    * If there someone listed as the dependent's parents and they are filing taxes, they will be assigned to their parent 
    * If there is not a parent, they will randomly be assigned to a tax-filing relative (not housemate or other non-relative) in the household 
- A simulant eligble to be a dependent must: 
    * Not be a "housemate/roommate" or "other nonrelative" to whoever is claiming them 
    * The dependent's income must be below $4300 
- For simulants outside of the household, they will be claimed as a dependent by their parent until age 19 OR age 24 if they are in group quarters for college 
    * For simulants born in the simulation, they should have at least one parent tracked 

>     * For simulants not born in the simulantion, information on initializing parents is being added shortly 

Reviewer: when you do add this info, if you end up deciding to track multiple parents, will you include info on how to decide which parent claims?

Author: Good point! I added something about this here.

    - 
  * - Age 
    -  
  * - Social Security Number or ITIN
    -

If a simulant does not have a SSN but is filing taxes, please 

> include an Individual Taxpayer Identification Number (ITIN) instead. 

Reviewer 2: Will you want this to be the same for a given simulant across years? If so, you might want to add a section about initializing it, separate from the tax observer, and just reference it here.

(Sorry, I initially put this comment on the commit instead of the PR.)

Author: Hmm we do want to track over time, but I am imagining that this is assigned on time steps, not initialized for anyone. It will just apply on as "as needed" basis. For that - should I still add it to another section? Or just here with some more written about how we want it tracked?

Reviewer 2: Hm... you mean that someone would only be assigned an ITIN when they are actually observed filing taxes? That makes sense but I am not sure if observers assigning values is something we do in Vivarium. That's why I had assumed it would be assigned upfront and just _used_ by the observer.

Author: Yeah, exactly, you just assign the number when the observer needs it. @beatrixkh can you weigh in? I can initialize this for everyone if needed.

Reviewer: sorry for the delay! yes, if someone should have the same ITN over time, we'll want to initialize it with them as zeb is suggesting. then if they never use it you'll never see it, but if they use it, it's permanently associated with them and won't change from one timestep to another.

and yes --- also, as zeb was thinking, observers are not able to effect the state of anything in the simulation (such as a simulant's id, age, etc.), they are only looking at the current state of the simulation, potentially doing some processing that doesn't change the state of the simulation, and then outputting those observations.

Author: Thanks! I am going to change this to be randomly assigned for now. If it proves to be important for PRL, we can add this as a new initialized concept later.
